contracts/StorageAccessible.sol

```diff
@@ -23,4 +23,41 @@ contract StorageAccessible {
         }
         return result;
     }
+
+    /**
+     * @dev Performs a delegetecall on a targetContract in the context of self.
+     * Internally reverts execution to avoid side effects (making it static). Catches revert and returns encoded result as bytes.
+     * @param targetContract Address of the contract containing the code to execute.
+     * @param calldataPayload Calldata that should be sent to the target contract (encoded method name and arguments).
+     */
+    function executeStaticDelegatecall(
+        address targetContract,
+        bytes memory calldataPayload
+    ) public returns (bytes memory) {
+        bytes memory innerCall = abi.encodeWithSignature(
+            "executeStaticDelegatecallInternal(address,bytes)",
```

Comment:
General question: is it more effective to hardcode the bytes, or does the compiler solve this? Might be worth considering it if this is only optimised when the optimiser is enabled.

Reply:
Good point. Will make this a constant

Reply:
Actually, since this call depends on calldata and targetContract I can't hardcode all of it. I can avoid doing the keccak hash an just append the arguments to it.